src/math/trig.c

```diff
@@ -10,93 +10,154 @@
-	int32_t b_idx;
+
+/* 652032874 , deg = 69.586061*/
+const int32_t cordic_sine_cos_lut_q29fl	 =  Q_CONVERT_FLOAT(1.214505869895220, 29);
+/* 1686629713, deg = 90.000000	*/
+const int32_t cordic_sine_cos_piovertwo_q30fl  = Q_CONVERT_FLOAT(_M_PI / 2, 30);
+/* 421657428 , deg = 90.000000 */
+const int32_t cord_sincos_piovertwo_q28fl  = Q_CONVERT_FLOAT(_M_PI / 2, 28);
+/* 843314857,  deg = 90.000000	*/
+const int32_t cord_sincos_piovertwo_q29fl  = Q_CONVERT_FLOAT(_M_PI / 2, 29);
+
+/**
+ * \Compute fixed point cordicsine with table lookup and interpolation
+ * \The cordic sine algorithm converges, when the angle is in the range
+ * \[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
+ * \pi/2 is added or subtracted from the angle until it is within the range
+ * \[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
+ * \has range in [-1.0 to 1.0]
+ * \+------------------+-----------------+--------+--------+
+ * \| thRadFxp	       | cdcsinth	 |thRadFxp|cdcsinth|
+ * \+----+-----+-------+----+----+-------+--------+--------+
+ * \|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * \+----+-----+-------+----+----+-------+--------+--------+
+ * \| 32 | 28  |  1    | 32 | 31 |   1	 | 4.28	  | 1.31   |
+ * \+------------------+-----------------+--------+--------+
+ */
+inline int32_t sin_fixed_32b(int32_t th_rad_fxp)
+{
+	int32_t sign;
 	int32_t b_yn;
 	int32_t xn;
+	int32_t th_cdc_fxp;
+	uint32_t flags = 0;
+
+	cordic_sin_cos(th_rad_fxp, flags, &sign, &b_yn, &xn, &th_cdc_fxp);
+	th_cdc_fxp = sign * b_yn;
+	/*convert Q2.30 to Q1.31 format*/
+	return sat_int32(Q_SHIFT_LEFT((int64_t)th_cdc_fxp, 30, 31));
+}
+
+/**
+ * \Compute fixed point cordic sine cos with table lookup and interpolation
+ * \The cordic sine cos algorithm converges, when the angle is in the range
+ * \[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
+ * \pi/2 is added or subtracted from the angle until it is within the range
+ * \[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
+ * \has range in [-1.0 to 1.0]
+ * \+------------------+-----------------+--------+------------+
+ * \| thRadFxp	       | cdcsincosth	 |thRadFxp| cdcsincosth|
+ * \+----+-----+-------+----+----+-------+--------+------------+
+ * \|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * \+----+-----+-------+----+----+-------+--------+------------+
+ * \| 32 | 28  |  1    | 32 | 15 |   1	 | 4.28	  | 1.15       |
+ * \+------------------+-----------------+--------+------------+
+ */
+
+inline int16_t sin_cos_fixed_16b(int32_t th_rad_fxp, uint32_t flags)
+{
+	int32_t sign;
+	int32_t b_yn;
+	int32_t xn;
+	int32_t th_cdc_fxp;
+
+	/* compute coeff from angles*/
+	cordic_sin_cos(th_rad_fxp, flags, &sign, &b_yn, &xn, &th_cdc_fxp);
+	/* sine */
+	if ((flags & EN_CORDIC_SINE) == EN_CORDIC_SINE)
+		th_cdc_fxp = sign * b_yn;
+	/* cosine */
+	if ((flags & EN_CORDIC_COSINE) == EN_CORDIC_COSINE)
```

Comment:
I assume there's never need to compute more than SINE or COSINE at time. The bitfield kind of suggests that both can be done but it doesn't look the case. Cosine "wins" if both are set. Would flags be better as enum type with values CORDIC_SINE, CORDIC_COSINE? Maybe a switch-case is suitable here (room to later add more like tangent function)?